packages/apollo-links/src/authHttpLink.ts

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020-2022 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import { from, ApolloLink, HttpOptions } from '@apollo/client/core';
+
+import { AuthLink } from './auth-link';
+import { DynamicHttpLink } from './dynamicHttpLink';
+import agreementMananger from './agreementMananger';
+import { creatErrorLink } from './errorLink';
+import { retryLink } from './retryLink';
+import { Logger } from "./types";
+
+interface AuthHttpOptions {
+  authUrl: string;          // auth service url
+  projectChainId: string;   // genesis hash of the chain
+  httpOptions: HttpOptions; // http options for init `HttpLink`
+  deploymentId: string;     // deployment id of the project
```

Comment:
Do we need `deploymentId` from sdk? I think it is out of the knowledge of sdk.

Reply:
Moved the auth link support multi-indexers out of authLink, it doesn't need deploymentId any more